test/test-cases/functional/saic/test_vm_to_vm_commn_udp_bidir.py

```diff
@@ -0,0 +1,299 @@
+ENI_VTEP_IP = "221.0.1.11"
+NETWORK_VTEP_IP = "221.0.2.101"
+
+OUTER_SRC_MAC = "80:09:02:01:00:01"
+OUTER_DST_MAC = "c8:2c:2b:00:d1:30" 
+INNER_SRC_MAC = "00:1A:C5:00:00:01"
+INNER_DST_MAC = "00:1b:6e:00:00:01"
+INNER_DST_MAC2= "00:1b:6e:00:00:03"
+OUTER_SRC_MAC_F2 = "80:09:02:02:00:01"
+OUTER_DST_MAC_F2 = "c8:2c:2b:00:d1:34"  
+
+###############################################################
+#                  Start of the testcase
+###############################################################
+
+
+@pytest.mark.xfail(reason="https://github.com/sonic-net/DASH/issues/345")
+class TestUdpBidir:
+
+    @pytest.fixture(scope="class")
+    def setup_config(self):
+        """
+        Fixture returns the content of the file with SAI configuration commands.
+        scope=class - The file is loaded once for the whole test class
+        """
+        current_file_dir = Path(__file__).parent
+        with (current_file_dir / 'config_bidir_setup_commands.json').open(mode='r') as config_file:
+            setup_commands = json.load(config_file)
+        return setup_commands
+
+    @pytest.mark.dependency()
+    def test_setup(self, dpu, setup_config):
+        results = [*dpu.process_commands(setup_config)]
+        print("\n======= SAI setup commands RETURN values =======")
+        pprint(results)
+        assert all(results), "Setup error"
+
+
+    @pytest.mark.dependency(depends=['TestUdpBidir::test_setup'])
+    def test_vm_to_vm_commn_udp_bidir(self, dataplane):
+
+        # Configure TGEN
+
+        print("\n======= Configure Flow1 from ENI to NETWORK =======")        
+        # Flow1 settings
+        f1 = dataplane.configuration.flows.flow(name="ENI_TO_NETWORK")[-1]
+        f1.tx_rx.port.tx_name = dataplane.configuration.ports[0].name
+        f1.tx_rx.port.rx_name = dataplane.configuration.ports[1].name
+        f1.size.fixed = PACKET_LENGTH
+        # send n packets and stop
+        f1.duration.fixed_packets.packets = TOTALPACKETS
+        # send n packets per second
+        f1.rate.pps = PPS
+        f1.metrics.enable = True
+
+        outer_eth1, ip1, udp1, vxlan1, inner_eth1, inner_ip1, inner_udp1= (
+                f1.packet.ethernet().ipv4().udp().vxlan().ethernet().ipv4().udp()
+        )
+
+        outer_eth1.src.value = OUTER_SRC_MAC
+        outer_eth1.dst.value = OUTER_DST_MAC
+        outer_eth1.ether_type.value = 2048
+
+        ip1.src.value = ENI_VTEP_IP #ENI - VTEP
+        ip1.dst.value = DPU_VTEP_IP #DPU - VTEP
+
+        udp1.src_port.value = 11638
+        udp1.dst_port.value = 4789
+
+        #vxlan.flags.value = 
+        vxlan1.vni.value = 11
+        vxlan1.reserved0.value = 0
+        vxlan1.reserved1.value = 0
+
+        inner_eth1.src.value = INNER_SRC_MAC
+        inner_eth1.dst.value = INNER_DST_MAC
+
+        inner_ip1.src.value = ENI_IP   #ENI
+        inner_ip1.dst.value = NETWORK_IP1  #world
+
+        inner_udp1.src_port.value = 10000
+        inner_udp1.dst_port.value = 20000
```

Comment:
Please consider using dataplane's methods add_flow(), add_ethernet_header(), add_*_header() or add add_vxlan_flow() function to your snappi_utils.py.
As I mentioned in my first comment, the test code is long and not easy to comprehend.